gridstatus/eia.py

```diff
@@ -149,6 +154,222 @@ def fetch_page_wrapper(url, headers, page, page_size):
+                        d1 = s1.find_next_sibling("td", class_="d1").text
+                        direction = float(
+                            s1.find_next_sibling("td", class_=directions).text,
+                        )
+                        df_petrol.loc[len(df_petrol)] = (
+                            text,
+                            s2,
+                            d1,
+                            float(direction) if direction != "NA" else "NA",
+                        )
+                    else:
+                        for i in range(rowspan_sum, rowspan + rowspan_sum):
+                            s2_elements = parent.select("td.s2")
+                            d1_elements = parent.select("td.d1")
+                            direction_elements = parent.find_all(class_=directions)
+                            df_petrol.loc[len(df_petrol)] = (
+                                text,
+                                s2_elements[i].text,
+                                d1_elements[i].text,
+                                float(direction_elements[i].text),
+                            )
+
+                    rowspan_sum += rowspan
+                except TypeError:
+                    s2 = s1.find_next_sibling("td", class_="s2").text
+                    d1 = s1.find_next_sibling("td", class_="d1").text
+                    direction = float(
+                        s1.find_next_sibling("td", class_=directions).text,
+                    )
+                    df_petrol.loc[len(df_petrol)] = (
+                        text,
+                        s2,
+                        d1,
+                        float(direction) if direction != "NA" else "NA",
+                    )
+
+            natural_gas_spots = soup.select_one(
+                "table[summary='Spot Natural Gas and Electric Power Prices']",
+            )
+
+            for s1 in natural_gas_spots.select("td.s1"):
+                price_siblings = s1.find_next_siblings("td", class_="d1")
+                direction_siblings = s1.find_next_siblings("td", class_=directions)
+                df_ng.loc[len(df_ng)] = (
+                    s1.text,
+                    price_siblings[0].text,
+                    float(direction_siblings[0].text)
+                    if direction_siblings[0].text != "NA"
+                    else "NA",
+                    price_siblings[1].text,
+                    float(direction_siblings[1].text)
+                    if direction_siblings[1].text != "NA"
+                    else "NA",
+                    price_siblings[2].text,
+                )
+
+        df_ng["date"] = pd.to_datetime(close_date)
+        df_petrol["date"] = pd.to_datetime(close_date)
+
+        df_ng = utils.move_cols_to_front(df_ng, cols_to_move=["date"])
+        df_petrol = utils.move_cols_to_front(df_petrol, cols_to_move=["date"])
+
+        d = {
+            "df_petrol": df_petrol,
```

Comment:
let's make these keys names as explicit as possible. no need to use df in the name since it's clear it's a dataframe. 

perhaps just `petroleum` and `natural_gas`